### PR TITLE
New: Animate with AI [ED-16191]

### PR DESCRIPTION
**Pull request by naorsabag**
## PR Checklist

- [x] The commit message follows our guidelines:  https://github.com/elementor/elementor/blob/master/.github/CONTRIBUTING.md

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Summary

This PR can be summarized in the following changelog entry:

* convert user textual requests to motion effects

## Description
An explanation of what is done in this PR

* convert user textual requests to motion effects

## Test instructions
This PR can be tested by following these steps:

*

## Quality assurance

- [x] I have tested this code to the best of my abilities
- [ ] I have added unittests to verify the code works as intended
- [ ] Docs have been added / updated (for bug fixes / features)

Fixes #

### CODE REVIEW

File: modules/ai/assets/js/editor/hooks/use-animation-prompt.js [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+		// First enabled property under scrolling_effects
+		result[ firstEnableControlKey ] = {
+			isParent: true,
+			label: getControlData( firstEnableControlKey ).label,
+			value: jsonObject.enabled[ firstEnableControlKey ],
+			tabs: 0,
+		};
+
+		// Process other properties
+		const processProps = ( props ) => {
```
Comment (DennisNerush) [inline modules/ai/assets/js/editor/hooks/use-animation-prompt.js:37]: call it "processOtherProps" and drop the comment

File: modules/ai/assets/js/editor/hooks/use-animation-prompt.js [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+		for ( const [ , propValue ] of Object.entries( jsonObject ) ) {
+			if ( 'object' === typeof propValue ) {
+				processProps( propValue );
+			}
+		}
+
+		return result;
+	};
+
+	const digestResult = ( jsonString ) => {
```
Comment (DennisNerush) [inline modules/ai/assets/js/editor/hooks/use-animation-prompt.js:151]: this whole file and this part in particular is pretty complex, how about some unit tests to test the logic?

**Review by DennisNerush — DISMISSED** [review]

File: modules/ai/assets/js/editor/module.js [hunk 3]
```diff
@@ -177,6 +177,41 @@ export default class Module extends elementorModules.editor.utils.Module {
 			};
 		}
 
+		if ( [ 'animation', 'hover_animation' ].includes( aiOptions.type ) ) {
+			const getControlValue = () => Object.values( elementor.widgetsCache.heading.controls )
```
Comment (matipojo) [inline modules/ai/assets/js/editor/module.js:181]: Why `heading`?

File: modules/ai/assets/js/editor/page-content.js [hunk 4]
```diff
@@ -220,6 +228,39 @@ const PageContent = (
 		);
 	}
 
+	if ( 'animation' === type || 'hover_animation' === type ) {
+		const onCloseAnimationDialog = () => {
+			setControlValue( originalControlValue );
+			elementor.documents.getCurrent().history.setActive( false );
```
Comment (matipojo) [inline modules/ai/assets/js/editor/page-content.js:234]: Why false?

Maybe use `restorePrevSettings`

File: modules/ai/assets/js/editor/pages/form-animation/index.js [hunk 5]
```diff
@@ -0,0 +1,191 @@
...
+	}, [ data ] );
+
+	useEffect( () => {
+		return () => {
+			restorePrevSettings();
+		};
+	}, [] );
+
+	const showPreview = ( result ) => {
+		const summery = Object.keys( result )
```
Comment (matipojo) [inline modules/ai/assets/js/editor/pages/form-animation/index.js:60]:
```suggestion
		const summary = Object.keys( result )
```

File: modules/ai/assets/js/editor/pages/form-animation/index.js [hunk 6]
```diff
@@ -0,0 +1,191 @@
...
+			.filter( ( key ) => result[ key ].label )
+			.map( ( key ) => {
+				const item = result[ key ];
+				const tabs = '&nbsp;'.repeat( item.tabs );
+				const bulletIcon = item.tabs > 0 ? '&#8226; ' : '';
+				// Const value = ! item.isParent && item.value && ! item.value.sizes?.length ? `<br/>&nbsp;&#8226;&nbsp;${ item.value.size ?? item.value }` : '';
```
Comment (matipojo) [inline modules/ai/assets/js/editor/pages/form-animation/index.js:66]:
```suggestion
```

File: modules/ai/assets/js/editor/pages/form-animation/index.js [hunk 7]
```diff
@@ -0,0 +1,191 @@
...
+							placeholder={ __( 'Describe the animation you want to create...', 'elementor' ) }
+							name="prompt"
+							value={ prompt }
+							color="secondary"
+							onChange={ ( event ) => setPrompt( event.target.value ) }
+						/>
+					</Box>
+
+					{ showSuggestions && <PromptSuggestions suggestions={ autocompleteItems } onSelect={ setPrompt }>
+						{ /* <PromptLibraryLink libraryLink={ 'https://go.elementor.com/ai-prompt-library-animation/' } />*/ }
```
Comment (matipojo) [inline modules/ai/assets/js/editor/pages/form-animation/index.js:128]:
```suggestion
```

File: modules/ai/module.php [hunk 8]
```diff
@@ -1323,6 +1395,35 @@ public function ajax_ai_get_product_image_unification( $data ): array {
 		];
 	}
 
+	public function ajax_ai_get_animation( $data ): array {
+		$app = $this->get_ai_app();
```
Comment (matipojo) [inline modules/ai/module.php:1399]:
```suggestion
		$data['editor_post_id'] = $data['payload']['postId'];
		$this->verify_upload_permissions( $data );
		
		$app = $this->get_ai_app();
```

**Review by matipojo — CHANGES_REQUESTED** [review]

**Review by matipojo — APPROVED** [review]

**Pull request merged**